matrix_sdk_crypto/src/machine.rs

```diff
@@ -932,10 +928,30 @@ impl OlmMachine {
             .get(&KeyAlgorithm::Ed25519)
             .ok_or(EventError::MissingSigningKey)?;
 
-        Ok((
-            serde_json::from_value::<EventJson<ToDeviceEvent>>(decrypted_json)?,
-            signing_key.to_owned(),
-        ))
+        let ev_type = decrypted_json
+            .get("type")
+            .cloned()
+            .ok_or_else(|| EventError::MissingField("type".to_string()))?;
+
+        let content = decrypted_json
+            .get("content")
+            .cloned()
+            .ok_or_else(|| EventError::MissingField("content".to_string()))?;
+
+        let sender = decrypted_json
+            .get("sender")
+            .cloned()
+            .ok_or_else(|| EventError::MissingField("sender".to_string()))?;
+
+        let device_event = json!({
+            "type": ev_type,
+            "content": content,
+            "sender": sender
+        });
+
+        let event = serde_json::from_value::<EventJson<AnyToDeviceEvent>>(device_event)?;
+
+        Ok((event, signing_key.to_owned()))
```

Comment:
The old `ToDevice` had a custom `Deserialize` impl so now that it is just derived it has to be picked apart.